### PR TITLE
Add logging of ftdi instruments

### PR DESCRIPTION
**Pull request by John2202W**
Add logging of FTDI instruments.
Allow substring matching to the ftdi description.
(more details - CPE2543)

### CODE REVIEW

File: src/fixate/drivers/ftdi.py [hunk 1]
```diff
@@ -450,3 +441,21 @@ def _serial_shift_bit_bang(self, data, bytes_required, bb_mask):
         data_out.append(bb_mask + self.bb_latch ^ self.bb_inv_mask)
         data_out.append(bb_mask + self.bb_inv_mask)
         return data_out
+
+    def get_identity(self) -> str:
+        """Return identity string representing connected ftdi object"""
+        return f"{self.ftdi_description.decode()}"
+
```
Comment (clint-lawrence) [inline src/fixate/drivers/ftdi.py:448]: `return self.ftdi_description.decode()`?

Comment (John2202W) [reply src/fixate/drivers/ftdi.py:448]: The ftdi_description comes from the raw FT Device info. Otherwise ends with the raw string as b'InstrumentX'

Comment (John2202W) [reply src/fixate/drivers/ftdi.py:448]: Yes, previously had some some other details in there but can fix that now

Comment (John2202W) [reply src/fixate/drivers/ftdi.py:448]: Done

**Comment by clint-lawrence** [comment]
I don't anticipate this being a problem, but I occurred to me that you can't prevent a substring matching. For example if we had two devices, "abc" and "abcd", and you really wanted to only open "abc", not "abcd" there is no way to specify that. I can't imagine a realistic scenario where that would come up.

Could use fnmatch to add wildcard matching or a regex if we need to.

**Comment by John2202W** [comment]
Have changed to regex to give more flexibility if/when needed. Still up to the test-script author to prevent unwanted substrings
e.g `ftdi.open(r"abc\-?\d? Some Description$")`

**Comment by clint-lawrence** [comment]
What about using `re.fullmatch` instead of `re.match`? With that the existing behavior of strict matching will be maintained, but allows more flexible matching going forward.

**Comment by John2202W** [comment]
Updated ^

**Comment by clint-lawrence** [comment]
Looks good to me

**Comment by jcollins1983** [comment]
Looks good. I can merge once the conflict on the release notes in resolved from the merge of PR #169

**Pull request merged**